aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/AWSAppSyncClient.java

```diff
@@ -444,6 +506,30 @@ public AWSAppSyncClient build() {
                     mServerUrl = appSyncJsonObject.getString("ApiUrl");
                     mRegion = Regions.fromName(appSyncJsonObject.getString("Region"));
 
+                    // Populate the ClientDatabasePrefix from awsconfiguration.json
+                    String clientDatabasePrefixFromConfigJson = null;
+                    try {
+                        clientDatabasePrefixFromConfigJson = appSyncJsonObject.getString("ClientDatabasePrefix");
+                    } catch (Exception ex) {
+                        Log.e(TAG, "Error is reading the ClientDatabasePrefix from AppSync configuration in awsconfiguration.json.");
+                        // Not throwing an exception for backwards compatibility reasons
+                        // where the awsconfiguration.json doesn't have ClientDatabasePrefix.
+                    }
+
+                    // ClientDatabasePrefix passed via code takes precedence over the one present in awsconfiguration.json
+                    mClientDatabasePrefix = mClientDatabasePrefix != null ? mClientDatabasePrefix : clientDatabasePrefixFromConfigJson;
+
+                    // If mUseClientDatabasePrefix is true, a valid prefix is required
+                    // Else do not use a prefix and use the default database names without a prefix
+                    if (mUseClientDatabasePrefix) {
+                        if (clientDatabasePrefixFromConfigJson == null) {
+                            throw new RuntimeException("ClientDatabasePrefix is not present in AppSync configuration in awsconfiguration.json " +
```

Comment:
This will crash the app - Can you look into throwing a checked exception. I understand if this being done for backward compatibility reasons.

Reply:
Since the `build()` throws RuntimeException and for consistency and backwards compatibility reasons, I am using RuntimeException as opposed to checked exception.

Reply:
sounds good.